libs/estdlib/src/base64.erl

```diff
@@ -0,0 +1,86 @@
+%%
+%% Copyright (c) dushin.net
+%% All rights reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%
+
+%%-----------------------------------------------------------------------------
+%% @doc An implementation of a subset of the Erlang/OTP base64 interface.
+%%
+%% This module is designed to be API-compatible with the Erlang/OTP base64 module,
+%% with the following exceptions:
+%%
+%% <ul>
+%%     <li>No support for decoding data with whitespace in base64 data</li>
+%%     <li>No support for mime decoding functions</li>
+%% </ul>
+%% @end
+%%-----------------------------------------------------------------------------
+-module(base64).
+
+-export([encode/1, encode_to_string/1, decode/1, decode_to_string/1]).
+
+%%-----------------------------------------------------------------------------
+%% @param   Data the data to encode
+%% @returns the base-64 data encoded, as a binary
+%% @doc     Base-64 encode a binary or string, outputting a binary.
+%% @end
+%%-----------------------------------------------------------------------------
+-spec encode(binary()|string()) -> binary().
+encode(Data) when is_binary(Data) ->
+    atomvm:b64_encode(Data);
```

Comment:
why we don't just call them base64:encode and base64:decode avoiding atomvm:b64_encode and atomvm:b64_decode? I don't see any benefit about having a wrapper.

Reply:
What would your recommendation be for handling the case where a string is provided as input?  Have separate code that traverses the string data?  Or convert the string to a binary?  It seems to me that the former adds unnecessary complexity (and has issues with O(n) traversal of the list, multiple times), and the latter is equivalent to doing the same in Erlang.

Reply:
Okay, I extended then if to take a binary or a list, and changed the nifs to use the `base64` module/namespace.  What is your feeling about the _to_string variants in the `base64` module?  Do you think these should be implemented in nif code, as well?  The advantage I can see would be avoiding the allocation of the binary in the process heap (it would just be a malloc for the duration of the function), which has advantages for memory-constrained applications.

Reply:
_to_string variants now implemented in NIF code, as well